src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -419,22 +446,67 @@ private void StartSend(HttpClient client, HttpRequestMessage message)
                     _deadlineTimer = new Timer(DeadlineExceeded, null, _timeout.Value, Timeout.InfiniteTimeSpan);
                 }
 
-                SendTask = SendAsync(client, message);
+                Log.StartingCall(Logger, Method.Type, request.RequestUri);
+                GrpcEventSource.Log.CallStart(Method.FullName);
+
+                var diagnosticSourceEnabled = 
+                    GrpcDiagnostics.DiagnosticListener.IsEnabled() &&
+                    GrpcDiagnostics.DiagnosticListener.IsEnabled(GrpcDiagnostics.ActivityName, request);
+
+                Activity? activity = null;
+
+                // Set activity if:
+                // 1. Diagnostic source is enabled
+                // 2. Logging is enabled
+                // 3. There is an existing activity (to enable activity propagation)
+                if (diagnosticSourceEnabled || Logger.IsEnabled(LogLevel.Critical) || Activity.Current != null)
+                {
+                    activity = new Activity(GrpcDiagnostics.ActivityName);
+                    activity.AddTag(GrpcDiagnostics.GrpcMethodTagName, Method.FullName);
+                    activity.Start();
+
+                    if (diagnosticSourceEnabled)
+                    {
+                        GrpcDiagnostics.DiagnosticListener.Write(GrpcDiagnostics.ActivityStartKey, request);
```

Comment:
please use anonymous  type  (check out comment for Stop())

Reply:
it could make sense to check if somebody wants Start event - they don't necessarily do as everything can be logged when Stop is called.